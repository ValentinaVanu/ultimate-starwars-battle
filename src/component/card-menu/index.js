export {
  CardMenu
} from './card-menu.component'
