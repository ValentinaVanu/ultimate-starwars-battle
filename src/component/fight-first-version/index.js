export {
  FirstVersion
} from './fight-first-version.component'
