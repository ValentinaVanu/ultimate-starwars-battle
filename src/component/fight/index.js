export {
  Fight
} from './fight.component'
