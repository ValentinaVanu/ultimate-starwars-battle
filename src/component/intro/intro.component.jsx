import React, { useEffect, useRef, useState } from 'react'

import starWarsSound from './mp3/starwarssong.mp3'
import * as SI from './intro.styled'
import { Button } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux'
import * as IA from '../../store/intro.action'
import VolumeUpIcon from '@material-ui/icons/VolumeUp'
import VolumeOffIcon from '@material-ui/icons/VolumeOff'
import SkipNextIcon from '@material-ui/icons/SkipNext'


const Intro = () => {
  const audioRef = useRef()
  const [muted, skipIntro] = useSelector(({ intro }) => ([
    intro.muted,
    intro.skipIntro,
  ]))
  const dispatch = useDispatch()
  // dispatch(IA.skipIntroAction(skipIntro))

  useEffect(
    () => {
      audioRef.current.volume = 1
      audioRef.current.muted = muted
      !muted && audioRef.current.play()
      console.log(audioRef.current)
      setTimeout(() => {
        dispatch(IA.skipIntroAction(true))
      }, 9999999920000);
    },
    [muted]
  )

  const toggleMute = () => {
    // setMuted(!muted)
    dispatch(IA.setMutedAction(!muted))
  }
  const handleSkipIntro = () => {
    dispatch(IA.skipIntroAction(true))
  }

  return (
    <>
      {!skipIntro && <SI.DeathStarBackgroud>

        {muted && <VolumeOffIcon
          style={{ color: "white", margin: '2rem' }}
          onClick={toggleMute}
        />}
        {!muted && <VolumeUpIcon
          style={{ color: "white", margin: '2rem' }}
          onClick={toggleMute}
        />}

        <SI.StyleCredits>
          <SI.StyledCrawl>
            <audio ref={audioRef} src={starWarsSound} />
            <h1>Ultimate StarWars Battle</h1>
            <p> Build an application to select random people or starships and render their details to see who would win based on a common attribute.</p>
            <p>A person with greater mass wins, a starship with more crew wins.</p>
            <p>Once two cards are displayed the app should declare one of the cards a winner based on the higher common attribute.</p>
            <p>Being Creative</p>
          </SI.StyledCrawl>
        </SI.StyleCredits>
        <SI.StyledButton
          onClick={handleSkipIntro}
        >
          Skip Intro <SkipNextIcon />
        </SI.StyledButton>
      </SI.DeathStarBackgroud>}
    </>
  )
}

export { Intro }
