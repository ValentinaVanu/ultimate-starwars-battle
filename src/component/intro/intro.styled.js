import { Button } from 'react-bootstrap'
import styled from 'styled-components'
import deathStar from './img/deathstar.jpg'



export const StyledButton = styled(Button)`
  && {
    background-color: #000;
    border-color: #7f723b;
    color: #cbcbcb;
    margin: 2rem;
    align-self: flex-end;

    :hover{
      /* background-color: rgba(48,48,51,0.93); */
      border-color: #feda4a;
    }
  }
`

export const StyleCredits = styled.div`
  overflow: hidden;
  display: flex;
  justify-content: center;
  position: relative;
  height: 100%;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: center;

  &::before {
    content: '';
    position: absolute;
    pointer-events: none;
    width: 100%;
    min-height: 60vh;
    top: -25px;
    background-image: linear-gradient(0deg, transparent, black 75%);
    z-index: 1;
  }
`

export const StyledCrawl = styled.div`
  position: relative;
  top: 999999px;
  transform-origin: 50% 100%;
  animation: crawl 50s linear;

  && h1 {
    font-size: 90%;
    text-align: center;
    margin: 0 0 100px
  }
  @keyframes crawl {
    0% {
      top: 200px;
      transform: rotateX(20deg)  translateZ(0);
    }
    100% { 
      top: -6000px;
      transform: rotateX(25deg) translateZ(-2500px);
    }
  }
`

export const DeathStarBackgroud = styled.div`
  display: flex;
  flex-direction: column;
  background-image: url(${deathStar});
  background-position: center center;
  background-size: cover;
  height: 100%;
`
