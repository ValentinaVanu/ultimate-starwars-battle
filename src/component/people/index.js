export {
  People
} from './people.component'
