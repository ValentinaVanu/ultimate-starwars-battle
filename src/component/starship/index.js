export {
  Starship
} from './starship.component'
