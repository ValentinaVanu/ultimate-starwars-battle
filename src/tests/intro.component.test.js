import { render } from '@testing-library/react'
import { Intro } from '../component/intro'
jest.mock("react-redux", () => ({
  useDispatch: f => f,
  useSelector: (f) => []
}))

describe("Intro component", () => {
  it("should render properly", () => {
    const component = render(<Intro />)
    expect(component).toMatchSnapshot()
  })
})
