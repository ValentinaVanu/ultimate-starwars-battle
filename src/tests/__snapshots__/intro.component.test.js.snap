// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hKgILt cQFPrf"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          style="color: white; margin: 2rem;"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
          />
        </svg>
        <div
          class="sc-gsTCUz gJwbhO"
        >
          <div
            class="sc-dlfnbm iHIaWA"
          >
            <audio
              src="starwarssong.mp3"
            />
            <h1>
              Ultimate StarWars Battle
            </h1>
            <p>
               Build an application to select random people or starships and render their details to see who would win based on a common attribute.
            </p>
            <p>
              A person with greater mass wins, a starship with more crew wins.
            </p>
            <p>
              Once two cards are displayed the app should declare one of the cards a winner based on the higher common attribute.
            </p>
            <p>
              Being Creative
            </p>
          </div>
        </div>
        <button
          class="sc-bdfBwQ cEuKaJ btn btn-primary"
          type="button"
        >
          Skip Intro 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hKgILt cQFPrf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="color: white; margin: 2rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
        />
      </svg>
      <div
        class="sc-gsTCUz gJwbhO"
      >
        <div
          class="sc-dlfnbm iHIaWA"
        >
          <audio
            src="starwarssong.mp3"
          />
          <h1>
            Ultimate StarWars Battle
          </h1>
          <p>
             Build an application to select random people or starships and render their details to see who would win based on a common attribute.
          </p>
          <p>
            A person with greater mass wins, a starship with more crew wins.
          </p>
          <p>
            Once two cards are displayed the app should declare one of the cards a winner based on the higher common attribute.
          </p>
          <p>
            Being Creative
          </p>
        </div>
      </div>
      <button
        class="sc-bdfBwQ cEuKaJ btn btn-primary"
        type="button"
      >
        Skip Intro 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 18l8.5-6L6 6v12zM16 6v12h2V6h-2z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
